cfme/infrastructure/host.py

```diff
@@ -558,3 +564,35 @@ def find_quadicon(host, do_not_navigate=False):
             return quadicon
     else:
         raise HostNotFound("Host '{}' not found in UI!".format(host))
+
+
+def register(provider, file_path, cancel=False):
+    """Register new nodes (Openstack)
+
+    Input file path of json file with new node details, navigation has to
+    be from a specific provider
+    """
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    tb.select('Configuration', 'Register Nodes')
+    my_form = {'file': file_path}
+    fill(register_nodes_form, my_form, action=register_nodes_form.register)
+
+
+def host_exist(provider, name='my_node'):
+    """" registered imported host exist
+    provider - provider instance
+    name - by default name is my_name
+    Input provider, name of the new node, looking for the host in Ironic
+    clients, compare the record found with hosts list in CFME DB
+    """
+    nodes = provider.mgmt.list_node()
+    nodes_dict = {i.name: i for i in nodes}
+    query = provider.appliance.db.session.query(
+        provider.appliance.db['hosts'], 'guid')
+    node_uuid = str(nodes_dict[name])
+    for db_node in query.all():
```

Comment:
@quarckster I am looking for a node c1910a38-dc96-11e6-a8d8-525400c4a564, in case I will use first() I will retrieve the wrong node from query

In [6]: query.all()
Out[6]:
[(<utils.db.hosts at 0xc83ec50>, u'f49794ca-d637-11e6-a8d8-525400c4a564'),
 (<utils.db.hosts at 0xc83ecd0>, u'f4b86812-d637-11e6-a8d8-525400c4a564'),
 (<utils.db.hosts at 0xc83ed50>, u'c1910a38-dc96-11e6-a8d8-525400c4a564'),
 (<utils.db.hosts at 0xc83edd0>, u'575838f6-d8a6-11e6-a8d8-525400c4a564'),
 (<utils.db.hosts at 0xc83ee50>, u'f4bfc120-d637-11e6-a8d8-525400c4a564'),
 (<utils.db.hosts at 0xc83eed0>, u'f4c6f972-d637-11e6-a8d8-525400c4a564')]

In [7]: query.first()
Out[7]: (<utils.db.hosts at 0xc83ec50>, u'f49794ca-d637-11e6-a8d8-525400c4a564')